Examples/ArcGISToolkitExamples/ARExample.swift

```diff
@@ -0,0 +1,102 @@
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ARKit
+import ArcGISToolkit
+import ArcGIS
+
+open class ARExample: UIViewController {
+    
+    public let arView = ArcGISARView(frame: .zero)
+
+    override open func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Example of how to get ARSessionDelegate methods from the ArcGISARView.
+        arView.sessionDelegate = self
+        arView.arSCNViewDelegate = self
+
+        view.addSubview(arView)
+        arView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            arView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            arView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            arView.topAnchor.constraint(equalTo: view.topAnchor),
+            arView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+            ])
+        
+        arView.sceneView.scene = makeStreetsScene()
+    }
+    
+    override open func viewDidAppear(_ animated: Bool) {
+        arView.startTracking()
+    }
+    
+    override open func viewDidDisappear(_ animated: Bool) {
+        arView.stopTracking()
+    }
+
+    private func makeStreetsScene() -> AGSScene {
+        
+        // create scene with the streets basemap
+        let scene = AGSScene(basemapType: .streets)
+        
+        // create elevation surface
+        let elevationSource = AGSArcGISTiledElevationSource(url: URL(string: "http://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer")!)
+        let surface = AGSSurface()
+        surface.elevationSources = [elevationSource]
+        surface.name = "baseSurface"
+        surface.isEnabled = true
+        surface.backgroundGrid.isVisible = false
+        scene.baseSurface = surface
+        
+        return scene
+    }
+}
+
+extension ARExample: ARSessionDelegate {
+    
+    public func session(_ session: ARSession, didUpdate frame: ARFrame) {
+        // Example of how to get ARSessionDelegate methods from the ArcGISARView.
+    }
+}
+
+extension ARExample: ARSCNViewDelegate {
+    
+    public func session(_ session: ARSession, didFailWithError error: Error) {
+        guard error is ARError else { return }
+        
+        let errorWithInfo = error as NSError
+        let messages = [
+            errorWithInfo.localizedDescription,
+            errorWithInfo.localizedFailureReason,
+            errorWithInfo.localizedRecoverySuggestion
+        ]
+        
+        // Remove optional error messages.
+        let errorMessage = messages.compactMap({ $0 }).joined(separator: "\n")
+        
+        DispatchQueue.main.async { [unowned self] in
```

Comment:
There's no guarantee that `self` will be around when this block is executed on the main queue. Therefore you need to weakly capture `self` here.

To prevent having to include multiple references to `self`, I usually extract the contents of a closure like this into a method, and then just delegate to that method include the closure. Something like this:

```swift
DispatchQueue.main.async { [weak self] in self?.showError(withMessage: errorMessage) }
```

Reply:
Got it, thanks.  Since there were only two references to self, I did not break the code out into a method.